### PR TITLE
Lower hardswish

### PR DESCRIPTION
**Pull request by lucasc896**
https://github.com/pytorch/xla/issues/3492

Counter tests are currently failing, but commented out. Would appreciate some guidance on fixing.

### CODE REVIEW

**Comment by lucasc896** [comment]
@JackCaoG It seems there's a problem with the gcloud service account auth in the docker image build:

```
ERROR: (gcloud.auth.activate-service-account) There was a problem refreshing your current auth tokens: ('invalid_grant: Invalid JWT Signature.', {'error': 'invalid_grant', 'error_description': 'Invalid JWT Signature.'})
```

**Comment by JackCaoG** [comment]
counter test are meant to make sure `xla` version of the op(the one you lowered) is called, if it is failing it is likely something went wrong.

**Comment by JackCaoG** [comment]
hmm our CI master is our today, @yeounoh Can you take a look when you have time?

**Comment by wonjoolee95** [comment]
> hmm our CI master is out today, @yeounoh Can you take a look when you have time?

Just FYI, I saw the same issue on https://github.com/pytorch/xla/pull/3497 but now it seems to be passing, so just retrying the CI might help!

File: torch_xla/csrc/elementwise.cpp [hunk 1]
```diff
@@ -92,6 +92,33 @@ xla::XlaOp BuildHardSigmoidBackward(xla::XlaOp grad_output, xla::XlaOp input) {
...
+  xla::XlaOp one = XlaHelpers::ScalarValue<float>(1.0, shape.element_type(),
+                                                  input.builder());
+  xla::XlaOp pointfive = XlaHelpers::ScalarValue<float>(
+      0.5, shape.element_type(), input.builder());
+
+  xla::XlaOp stepone =
+      xla::Select(Between(input, -3.0, 3.0),
+                  xla::Mul(grad_output, pointfive + (input / three)), zero);
+
+  return xla::Select(xla::Ge(input, three), one, stepone);
```
Comment (miladm) [inline torch_xla/csrc/elementwise.cpp:119]: The derivative for `x>=3` seems to be `1 * grad_output`; right?

Looks like this line should be:

```
return xla::Select(xla::Ge(input, three), grad_output, stepone);
```

Then, we won't need these [lines](https://github.com/pytorch/xla/pull/3496/files#diff-777be0b41499d13e65c42f07f961ee5810a4cd79e66665fa8da1519edcd8b31aR110-R111).

**Review by miladm — CHANGES_REQUESTED** [review]
Thanks for your contribution @lucasc896!

I added a couple of comments below.

File: test/cpp/test_aten_xla_tensor.cpp [hunk 2]
```diff
@@ -5786,6 +5786,46 @@ TEST_F(AtenXlaTensorTest, TestHardSigmoidBackward) {
...
+    return torch::hardswish(inputs[0]);
+  };
+  ForEachDevice([&](const torch::Device& device) {
+    TestBackward(
+        {torch::randn({10},
+                      torch::TensorOptions(torch::kFloat).requires_grad(true))},
+        device, testfn);
+  });
+
+  ExpectCounterNotChanged("aten::.*", cpp_test::GetIgnoredCounters());
```
[the remark this answers is not in the document]
Comment (lucasc896) [reply test/cpp/test_aten_xla_tensor.cpp:5826]: Done. I notice the above test for `hardsigmoid_backwared` similarly doesn't have this specific counter. Shall I add it to this PR?

File: torch_xla/csrc/elementwise.cpp (hunk 1 above)
Comment (lucasc896) [reply torch_xla/csrc/elementwise.cpp:119]: Great spot! Thanks for double checking this.

**Comment by lucasc896** [comment]
Looks like CI is unhappy again:

```bash
Cloning into '/tmp/pytorch/third_party/eigen'...
error: RPC failed; HTTP 500 curl 22 The requested URL returned error: 500 Internal Server Error
fatal: The remote end hung up unexpectedly
fatal: clone of 'https://gitlab.com/libeigen/eigen.git' into submodule path '/tmp/pytorch/third_party/eigen' failed
Failed to clone 'third_party/eigen'. Retry scheduled
```

I've had this twice now. Any suggestions, other than re-triggering again later today?

File: test/cpp/test_aten_xla_tensor.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (miladm) [reply test/cpp/test_aten_xla_tensor.cpp:5826]: Yes, feel free to add it there too. Thank you!

**Comment by miladm** [comment]
CircleCI seems to have a bad day. Trigger it once more today. It should go through.

**Review by miladm — APPROVED** [review]
Thanks @lucasc896 

Let's add the test line for sigmoid as you suggested and have the tests pass. Otherwise, LGTM.

**Comment by JackCaoG** [comment]
@miladm Does this change lgty?

**Comment by miladm** [comment]
All look good. Thanks @lucasc896. Merging.
